dll/win32/advapi32/reg/reg.c

```diff
@@ -1108,6 +1108,12 @@ RegCreateKeyExW(
 
     TRACE("RegCreateKeyExW() called\n");
 
+    if (lpSecurityAttributes && lpSecurityAttributes->nLength == 0)
```

Comment:
Maybe the question is whether the **other** check (line 1117 that returns ERROR_INVALID_USER_BUFFER) is actually correct.
(The reason I'm asking is that it would appear that's the only function that does such check, while all others just expect the kernel to do the verification.)
Note that this was added in commit https://github.com/reactos/reactos/commit/cc0716b65d18871f3124a02dcbb2c717a6c38c96 .

Reply:
See new tests in [CORE-15471](https://jira.reactos.org/browse/CORE-15471). Seems to prove that this is correct.

Reply:
@HBelusca, it seems that @GeoB99 was able to make me understand what your original idea was here about whether the __other__ check was actually correct. See what you think of the latest changes. It passes my new test as well. Thanks.